pkg/distributor/distributor.go

```diff
@@ -482,7 +482,7 @@ func (d *Distributor) createShards(stream logproto.Stream, totalShards int, tena
 	startShard := d.shardTracker.LastShardNum(tenantID, stream.Hash)
 	for i := 0; i < streamCount; i++ {
 		shardNum := (startShard + i) % totalShards
-		shard := d.createShard(streamLabels, streamPattern, shardNum)
+		shard := d.createShard(streamLabels, streamPattern, shardNum, len(stream.Entries)/totalShards)
```

Comment:
Whenever dividing by a number, you should always check that `int > 0`

Reply:
You're talking about a _loose_ check here because the check is done elsewhere before this code is run, right?
If we refactor the code later, that loose association might be broken.

In cases like this where a `panic` could result, it can't hurt to be extra careful.

Reply:
I pushed a commit adding the check to `createShards` (to not check if for every call to `createShard`) and logging an error if the problem is spotted. WDYT?

Reply:
I could also just do `max(totalShards, 1)` but I'd really like to log an error in such scenario because totalShards=0 shouldn't ever happen so logging it will be useful spotting a bigger problem